src/mlpack/methods/det/dtree_impl.hpp

```diff
@@ -151,8 +149,7 @@ void ExtractSplits(std::vector<std::pair<ElemType, size_t>>& splitVec,
 
 template<typename MatType, typename TagType>
 DTree<MatType, TagType>::DTree() :
-    start(0),
-    end(0),
+     end(0),
```

Comment:
Does `start` need to be initialized still too?  This could be the build failure issue, not sure.